File: app/src/main/java/com/tronography/pixelweather/PixelWeatherApplication.java
package com.tronography.pixelweather;

import android.app.Application;

import com.tronography.pixelweather.dagger.AppComponent;
import com.tronography.pixelweather.dagger.AppModule;
import com.tronography.pixelweather.dagger.DaggerAppComponent;
import com.tronography.pixelweather.dagger.NetworkModule;
import com.tronography.pixelweather.dagger.SharedPrefModule;


public class PixelWeatherApplication extends Application {

    private AppComponent component;

    @Override
    public void onCreate() {
        super.onCreate();
        component = createComponent();
    }

    public AppComponent getAppComponent() {
        return component;
    }

    public AppComponent createComponent() {
        return DaggerAppComponent.builder()
                .appModule(new AppModule(this))
                .networkModule(new NetworkModule())
                .sharedPrefModule(new SharedPrefModule())
                .build();
    }
}

File: app/src/main/java/com/tronography/pixelweather/PixelWeatherApplication.kt
package com.tronography.pixelweather

import android.app.Application

import com.tronography.pixelweather.dagger.AppComponent
import com.tronography.pixelweather.dagger.AppModule
import com.tronography.pixelweather.dagger.DaggerAppComponent
import com.tronography.pixelweather.dagger.NetworkModule
import com.tronography.pixelweather.dagger.SharedPrefModule


class PixelWeatherApplication : Application() {

    var appComponent: AppComponent? = null
        private set

    override fun onCreate() {
        super.onCreate()
        appComponent = createComponent()
    }

    fun createComponent(): AppComponent {
        return DaggerAppComponent.builder()
                .appModule(AppModule(this))
                .networkModule(NetworkModule())
                .sharedPrefModule(SharedPrefModule())
                .build()
    }
}
